ceph_iscsi_config/utils.py

```diff
@@ -86,7 +86,11 @@ def resolve_ip_addresses(addr):
         infos = socket.getaddrinfo(addr, 0)
         for info in infos:
             if info[0] in families:
-                addrs.add(info[4][0])
+                # remove interface scope suffix and IPv4-over-IPv6 prefix
+                addr = info[4][0]
+                addr = addr.rsplit('%', 1)[0]
+                addr = addr.split('::ffff:', 1)[-1]
```

Comment:
I think we want to skip any that start w/ '::ffff' (not just contains '::')

Reply:
I have not actually ever seen it returned. I only did that to match what requires_restricted_auth was doing:

requires_restricted_auth gets "::ffff:192.168.1.100" in request.remote_addr. That function drops the ::ffff, and then socket.getaddrinfo always returns "192.168.1.100".

I just added it above just in case there was a case getaddrinfo could return  "::ffff:192.168.1.100" instead of "192.168.1.100" and we just have not seen it.

Reply:
Gotcha. Yeah, those IPv4-over-IPv6 addresses technically need to start with '::ffff:' and not just a random `::ffff` anywhere in the address (i.e. since `1234::ffff:abcd` isn't an IPv4-over-IPv6 address)

Reply:
Ah I see what you are saying. I misunderstood how split() works. I can fix that up.

I was thinking why are you saying the split() use is wrong, because I thought you wrote it and I am just copying you :)

Reply:
Lol -- nope, I was wrong and it would inadvertently break any IPv6 addresses with an embedded `::ffff`